ops/services/alerts/app_service_metrics/main.tf

```diff
@@ -283,3 +283,32 @@ requests
     threshold = 0
   }
 }
+
+resource "azurerm_monitor_scheduled_query_rules_alert" "report_stream_uploader_400" {
+  name                = "${var.env}-report_stream_uploader_400"
+  description         = "${local.env_title} alert when the publish function results in a 400"
+  location            = data.azurerm_resource_group.app.location
+  resource_group_name = var.rg_name
+
+  action {
+    action_group = var.action_group_ids
+  }
+
+  data_source_id = var.app_insights_id
+  enabled        = contains(var.disabled_alerts, "report_stream_uploader_400") ? false : true
+
+  query = <<-QUERY
+customEvents
+| order by timestamp desc
+| extend httpStatus = toint(customDimensions["status"])
+| where httpStatus == 400
+  QUERY
+
+  severity    = 1
+  frequency   = 5
+  time_window = 6
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 0
+  }
+}
```

Comment:
if there's ever a 400 response and we don't get something in the file error queue for some reason, would we still want to know about it? or is it only really useful if/when manual intervention is required? might the 400 error already be covered by our other alert for the function failing?

Reply:
The way things are written now a 400 would always result in something in the file error queue.

So before, the 400 requests surfaced in the form of the azure function not running (whatever we call that alert).  This code change would make that same behavior silent from an alerting perspective (because the function will keep running successfully).

Reply:
ok yeah I think previously we'd trigger [this alert](https://github.com/CDCgov/prime-simplereport/blob/cbf1d9bf5954e10ebf500f3c1973df9c9c9f395f/ops/services/alerts/app_service_metrics/main.tf#L198) because the function would repeatedly fail on the same message but now it would only fail once when we [throw the error](https://github.com/CDCgov/prime-simplereport/blob/cbf1d9bf5954e10ebf500f3c1973df9c9c9f395f/ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/index.ts#L124) the first time, which wouldn't cross the alert threshold. which seems like a good thing
in which case yeah it makes sense to maybe just alert if something is on the failed queue

Reply:
I kind of came full circle on the alerting on the failed queue idea.  If this alert goes off in the middle of the night - its the sort of thing that can wait until the morning to fix (my 2 cents).  But if we alert on the queue we'd have to disable that alert to be able to go back to bed.